ci/ray_ci/macos/macos_ci.sh

```diff
@@ -11,31 +11,52 @@ export LANG="en_US.UTF-8"
 export BUILD="1"
 export DL="1"
 
+filter_flaky_tests() {
+  # This function takes in a test targets list file, filters out flaky tests and
+  # overwrites the original test targets list file with the filtered list.
+
+  test_targets_file="$1"
+  # temp file is needed because python script runs in bazel sandbox
+  temp_path="$(bazel info bazel-genfiles)/temp_file.txt"
+  # copy test target file content to temp file
+  cp "$test_targets_file" "$temp_path"
+  bazel run ci/ray_ci/automation:filter_tests -- "$temp_path" "flaky"
+  # copy temp file content back to test target file
+  cp "$temp_path" "$test_targets_file"
+}
+
 run_small_test() {
-  # shellcheck disable=SC2046
-  bazel test $(./ci/run/bazel_export_options) --config=ci \
+  bazel query 'attr(tags, "client_tests|small_size_python_tests", tests(//python/ray/tests/...))' > ci/ray_ci/macos/small_test_targets.txt
```

Comment:
can we save the output in a file (or dir) under `/tmp`, so that it is not polluting the source dir?

we should generally treat source directory as read only.

or just drop the file usage, and use pipes for everything, as you are already using `xargs`.